casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -397,6 +399,36 @@ object Validate {
     } yield status
   }
 
+  /*
+   * This check must come before Validate.parents
+   */
+  def justificationFollows[F[_]: Monad: Log: BlockStore](
+      b: BlockMessage,
+      genesis: BlockMessage,
+      dag: BlockDag): F[Either[InvalidBlock, ValidBlock]] =
+    for {
+      latestMessages <- ProtoUtil.toLatestMessage[F](b.justifications, dag)
+      validators     = latestMessages.keySet
+      creatorLatestBlock <- ProtoUtil.creatorJustification(b).foldM(genesis) {
```

Comment:
Please refactor the ProtoUtil.creatorJustification: we now store the latest messages in memory so you shouldn't have to fetch the creator justification.

Reply:
Done